### ISSUE TITLE
Unable to load livechat since updating to 3.3.0 due to X-Frame-Options: sameorigin

### ISSUE THREAD

**Issue by jakguru** [RocketChat/Rocket.Chat#17786]
### Description:

Today I updated my server from 2.9.x to 3.3.0. After finishing the upgrade, I navigated to one of the websites which utilizes the Live Chat (now Omnichat) integration, only to find a grey square (of an iframe which didn't load) where the Live Chat button should have been. I looked in the console and found an error caused by the `X-Frame-Options: sameorigin` header.

I tried specifically adding the hostname of the website(s) which utilize the Livechat, but that did not make any difference. I confirmed that the issue was caused by Rocket Chat and not by an intermediary by running a `curl` request directly from the server:

```
curl -v http://localhost:3000/livechat
```

Which responded with:

```
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 3000 (#0)
> GET /livechat HTTP/1.1
> Host: localhost:3000
> User-Agent: curl/7.58.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< X-XSS-Protection: 1
< X-Frame-Options: sameorigin
< X-Instance-ID: mpXqru2q2qES9pnw7
< content-type: text/html; charset=utf-8
< Vary: Accept-Encoding
< Date: Fri, 29 May 2020 20:03:59 GMT
< Connection: keep-alive
< Transfer-Encoding: chunked
```

**Note:** I have been able to mitigate the issue through the use of Cloudflare's Workers, which allows me to remove the header. This does resolve the issue, however I would prefer to not have to pay the $5/month to ensure that my self-hosted livechat is working correctly.

### Steps to reproduce:

1. From the machine hosting Rocket Chat, make a verbose curl request:

```
curl -v http://localhost:3000/livechat
```

### Expected behavior:

If there are domains configured, then the `X-Frame-Options` should reflect those domains. Otherwise it should not be present

### Actual behavior:

`X-Frame-Options: sameorigin` is returned

### Server Setup Information:

- Version of Rocket.Chat Server: 3.3.0
- Operating System: Linux
- Deployment Method: DigitalOcean
- Number of Running Instances: 1
- DB Replicaset Oplog: Enabled
- NodeJS Version: v12.16.1
- MongoDB Version:

**Comment by renatobecker** [comment]
@jakguru let me get this straight..
So, you're trying to reach the URL -> http://localhost:3000/livechat from one of your websites, is that correct?

Also, have you set up the following setting?
![Screen Shot 2020-05-29 at 17 55 51](https://user-images.githubusercontent.com/59577424/83304954-2f7cc200-a1d6-11ea-84c1-f4868389199b.png)

I'll be waiting for your feedback.

**Comment by jakguru** [comment]
> @jakguru let me get this straight..
> So, you're trying to reach the URL -> http://localhost:3000/livechat from one of your websites, is that correct?
> 
> Also, have you set up the following setting?
> ![Screen Shot 2020-05-29 at 17 55 51](https://user-images.githubusercontent.com/59577424/83304954-2f7cc200-a1d6-11ea-84c1-f4868389199b.png)
> 
> I'll be waiting for your feedback.

No, that is not correct. I'm trying to reach https://{domain}/livechat from one of my websites, and i'm getting an error on the console about being unable to load an iframe due to the `X-Frame-Options` header returning a value of `sameorigin`.

In order to ensure that the issue was caused by the application and not by any intermediary service such as a reverse proxy, i ran the `curl` request from the command line of the server which hosts the application.

**Comment by jakguru** [comment]
> @jakguru let me get this straight..
> So, you're trying to reach the URL -> http://localhost:3000/livechat from one of your websites, is that correct?
> 
> Also, have you set up the following setting?
> ![Screen Shot 2020-05-29 at 17 55 51](https://user-images.githubusercontent.com/59577424/83304954-2f7cc200-a1d6-11ea-84c1-f4868389199b.png)
> 
> I'll be waiting for your feedback.

Also, I've tried the "Livechat Allowed Domains" option and it didn't change anything, even after clearing cache and cookies in the browser (or just making a fresh `curl` request).

**Comment by renatobecker** [comment]
It seems like the setting below is impacting on your environment:

<img width="985" alt="Screen Shot 2020-05-29 at 22 48 20" src="https://user-images.githubusercontent.com/59577424/83316825-a9c13c80-a1fe-11ea-9c69-4372175871ee.png">

**Comment by jakguru** [comment]
> It seems like the setting below is impacting on your environment:
> 
> <img alt="Screen Shot 2020-05-29 at 22 48 20" width="985" src="https://user-images.githubusercontent.com/59577424/83316825-a9c13c80-a1fe-11ea-9c69-4372175871ee.png">

Shouldn't those settings be ignored for the livechat, since it MUST be loaded from within an iFrame?

**Comment by tboulogne** [comment]
Hello, unable to lord live chat to but with another message :
```
Refused to load https://mydomain/livechat because it does not appear in the frame-ancestors directive of the Content Security Policy.
```

I set this headers on the proxy (Caddy):

```
-Server
X-Frame-Options DENY
Strict-Transport-Security "max-age=31536000; includeSubdomains"
X-Content-Type-Options nosniff
Referrer-Policy no-referrer-when-downgrade
Content-Security-Policy "frame-src  https://mydomain/livechat"
Content-Security-Policy "frame-ancestors 'self' https://mydomainlivechat"
Feature-Policy "vibrate 'self'; sync-xhr 'self' mydomain"
```

Thanks for any tips to solve this.

regards,

**Comment by usamamashkoor** [comment]
@jakguru Did you find any solution for this, I am facing the same issue here?

**Comment by dlecan** [comment]
thank you @pazande, I had the same issue, but only on Brave browser (see https://github.com/RocketChat/Rocket.Chat.Livechat/issues/531).
I have disabled the option "Restrict access inside any Iframe" and have refreshed the page with my widget and it works fine now.

**Comment by y377** [comment]
First you need to determine your Web server type and add the correct headers.
For example, in Nginx, the correct header is: 'add_header x-frame-options https://www.ssl360.cn'

**Comment by y377** [comment]
Refer to the consequences of punctuating[bug](https://github.com/RocketChat/Rocket.Chat/issues/19695#issue-752313173)

**Comment by pawelbura** [comment]
> @jakguru let me get this straight.. So, you're trying to reach the URL -> http://localhost:3000/livechat from one of your websites, is that correct?
> 
> Also, have you set up the following setting? ![Screen Shot 2020-05-29 at 17 55 51](https://user-images.githubusercontent.com/59577424/83304954-2f7cc200-a1d6-11ea-84c1-f4868389199b.png)
> 
> I'll be waiting for your feedback.

Thanks! It works for me.
For me it seems, that it works a bit different than in this description: if X-Frame-Option is set to sameorigin, empty "Livechat Allowed Domains" blocks chat on any domain.

So my working setup is: X-Frame-Option enabled and set to sameorigin + "Livechat Allowed Domains" set to domain name (without "http://", just name).

And it works!
